### PR TITLE
feat(arc-203): limit metadata collection

### PR DESCRIPTION
**Pull request by bartnaessens**
- Implement permission checks on media queries
- return a limited set of fields on export collection
CleanShot 2022-04-26 at 14 35 29

### CODE REVIEW

File: src/modules/media/services/media.service.ts [hunk 1]
```diff
@@ -167,6 +167,22 @@ export class MediaService {
 		return object;
 	}
 
+	public adaptLimitedMetadata(graphQlObject: GqlLimitedIeObject): Partial<Media> {
```
Comment (bertyhell) [inline src/modules/media/services/media.service.ts:170]: this is not related to the limited vs extended metadata set, right?
https://meemoo.atlassian.net/browse/ARC-670

**Review by bertyhell — CHANGES_REQUESTED** [review]

File: src/modules/media/controllers/media.controller.ts [hunk 2]
```diff
@@ -119,8 +145,19 @@ export class MediaController {
...
-		@Param('meemooIdentifier') meemooIdentifier: string
+		@Param('meemooIdentifier') meemooIdentifier: string,
+		@SessionUser() user: SessionUserEntity
 	): Promise<any> {
+		// Check if the user can search in all index (meemoo admin)
+		const canSearchInAllSpaces = user.has(Permission.SEARCH_ALL_OBJECTS);
+
+		if (
+			!canSearchInAllSpaces &&
+			!(await this.userHasAccessToVisitorSpaceOrId(user, maintainerId))
```
[the remark this answers is not in the document]
Comment (bartnaessens) [reply src/modules/media/controllers/media.controller.ts:156]: not applicable here (related call, maintainerId is set)

File: src/modules/media/controllers/media.controller.ts [hunk 3]
```diff
@@ -100,6 +113,19 @@ export class MediaController {
 	): Promise<string> {
 		const objectMetadata = await this.mediaService.findMetadataBySchemaIdentifier(id);
 
+		// Check if the user can search in all index (meemoo admin)
+		const canSearchInAllSpaces = user.has(Permission.SEARCH_ALL_OBJECTS);
+
+		if (
+			!canSearchInAllSpaces &&
+			objectMetadata.maintainerId &&
+			!(await this.userHasAccessToVisitorSpaceOrId(user, objectMetadata.maintainerId))
```
[the remark this answers is not in the document]
Comment (bartnaessens) [reply src/modules/media/controllers/media.controller.ts]: updated 👍

**Review by bertyhell — APPROVED** [review]
